google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStub.java

```diff
@@ -120,65 +123,89 @@
 
   public static EnhancedBigtableStub create(EnhancedBigtableStubSettings settings)
       throws IOException {
-    ClientContext clientContext = ClientContext.create(settings);
+    settings = finalizeSettings(settings, Tags.getTagger(), Stats.getStatsRecorder());
 
-    return new EnhancedBigtableStub(
-        settings, clientContext, Tags.getTagger(), Stats.getStatsRecorder());
+    return new EnhancedBigtableStub(settings, ClientContext.create(settings));
   }
 
-  @InternalApi("Visible for testing")
-  public EnhancedBigtableStub(
-      EnhancedBigtableStubSettings settings,
-      ClientContext clientContext,
-      Tagger tagger,
-      StatsRecorder statsRecorder) {
-    this.settings = settings;
+  public static EnhancedBigtableStubSettings finalizeSettings(
+      EnhancedBigtableStubSettings settings, Tagger tagger, StatsRecorder stats)
+      throws IOException {
+    EnhancedBigtableStubSettings.Builder builder = settings.toBuilder();
+
+    if (settings.isRefreshingChannel()) {
+      // Fix the credentials so that they can be shared
+      Credentials credentials = null;
+      if (settings.getCredentialsProvider() != null) {
+        credentials = settings.getCredentialsProvider().getCredentials();
+      }
+      builder.setCredentialsProvider(FixedCredentialsProvider.create(credentials));
+
+      // Inject the primer
+      InstantiatingGrpcChannelProvider transportProvider =
+          (InstantiatingGrpcChannelProvider) settings.getTransportChannelProvider();
+
+      builder.setTransportChannelProvider(
+          transportProvider
+              .toBuilder()
+              .setChannelPrimer(
```

Comment:
It would be nice to have a comment here explaining the consequence of "setChannelPrimer" that the channel is created once, but periodically creates a new channel, runs BigtableChannelPrimer on the newly created channel and replaces the existing one with the new one.

Reply:
I dont quite follow, can you rephrase?